### PR TITLE
Support nested clones; add sprite.andClones()

### LINKED ISSUE #28: Support nested clones
To match Scratch behavior, creating a clone of a clone currently does a sneaky bait-and-switch where the original sprite is cloned again instead.

As [pointed out](https://github.com/PullJosh/scratch-js/issues/25#issuecomment-569548874) by @nanaian, this is not very friendly to the honorable individuals who choose to use scratch-js.

A better option would be to store clone instances in a properly nested manner, but update sb-edit, the Scratch project translator, to always create a clone of the original when `create clone of [myself v]` is invoked. That would look like `this.sprites.sprite1.createClone()`.

### PR DESCRIPTION
**Pull request by PullJosh**
Do the obvious correct thing instead of the non-obvious incorrect one.

Fixes #28

### CODE REVIEW

**Comment by PullJosh** [comment]
Anybody want to look this over before I merge? @nanaian @towerofnix

File: scratch-js/Sprite.mjs [hunk 1]
```diff
@@ -187,24 +187,15 @@ export class Sprite extends SpriteBase {
...
-        new Trigger(
-          trigger.trigger,
-          trigger.options,
-          trigger._script.bind(clone)
-        )
-    );
-    clone.costumes = parent.costumes;
-    clone._vars = Object.assign({}, parent._vars);
+    clone._project = this._project;
+    clone.triggers = this.triggers;
```
Comment (adroitwhiz) [inline scratch-js/Sprite.mjs]: Is it alright that we're not cloning the triggers anymore?

Comment (PullJosh) [reply scratch-js/Sprite.mjs]: I believe so. The method attached to the trigger is specifically **not** bound to a particular instance (which is the opposite of how it was before clones were introduced). The method being called is bound with the correct value of `this` *when the trigger fires*, which means that copying the triggers array directly from another sprite shouldn't cause any problems. In theory, anyway.

Comment (adroitwhiz) [reply scratch-js/Sprite.mjs]: It looks like the trigger's internal state (e.g. generator state) will be shared between the parent and clone, though. If you call `start()` on the parent trigger, won't the child trigger start over from the beginning too?

Comment (PullJosh) [reply scratch-js/Sprite.mjs]: Oh! I think you're right. Will change.

Comment (PullJosh) [reply scratch-js/Sprite.mjs]: How does the new code look?

Comment (adroitwhiz) [reply scratch-js/Sprite.mjs]: 👍

File: scratch-js/Sprite.mjs [hunk 2]
```diff
@@ -187,24 +187,17 @@ export class Sprite extends SpriteBase {
...
-    clone.triggers = parent.triggers.map(
-      trigger =>
-        new Trigger(
-          trigger.trigger,
-          trigger.options,
-          trigger._script.bind(clone)
-        )
+    clone._project = this._project;
+    clone.triggers = this.triggers.map(
+      trigger => new Trigger(trigger.trigger, trigger.options, trigger._script)
```
Comment (towerofnix) [inline scratch-js/Sprite.mjs]: this is already merged obviously (and an unrelated change to this pr) but should there be a method `trigger.clone()`? might be handy for simplifying stuff like this, getting rid of the need to assume the function signature!

Comment (PullJosh) [reply scratch-js/Sprite.mjs]: Yes. You wanna make a PR? :P

**Pull request merged**